src/Compilers/Core/Portable/Operations/OperationNodes.cs

```diff
@@ -5872,13 +5873,15 @@ public override void Accept(OperationVisitor visitor)
     /// </summary>
     internal sealed partial class ReturnOperation : BaseReturnOperation, IReturnOperation
     {
-        public ReturnOperation(OperationKind kind, IOperation returnedValue, SemanticModel semanticModel, SyntaxNode syntax, ITypeSymbol type, Optional<object> constantValue, bool isImplicit) :
+        public ReturnOperation(OperationKind kind, IOperation returnedValue, RefKind refKind, SemanticModel semanticModel, SyntaxNode syntax, ITypeSymbol type, Optional<object> constantValue, bool isImplicit) :
             base(kind, semanticModel, syntax, type, constantValue, isImplicit)
         {
             ReturnedValue = SetParentOperation(returnedValue, this);
+            RefKind = refKind;
         }
 
         public override IOperation ReturnedValue { get; }
+        public override RefKind RefKind { get; }
```

Comment:
>public override RefKind RefKind { get; } [](start = 8, length = 40)

Consider moving implementation to the base and share it for all derived types.

Reply:
@sharwell this is the general pattern that we use for any non-IOperation properties.

---
In reply to: [292230935](https://github.com/dotnet/roslyn/pull/36271#discussion_r292230935) [](ancestors = 292230935)

Reply:
Done in ff7f6eabaf5acd84e1b64eef4c01db273c5ac101